lib/task.js

```diff
@@ -384,22 +384,34 @@ function factory(
     };
 
     Task.prototype.cancel = function(error) {
+        var self = this;
         error = error || new Errors.TaskCancellationError();
-        this.error = error;
+        self.error = error;
 
-        if (_.contains(Constants.Task.FinishedStates, this.state)) {
-            return;
-        } else if (this.state === TaskStates.Pending) {
-            // This block handles cancellation before run() has
-            // been called (i.e. the task has been created, but not
-            // yet scheduled to run by the scheduler via taskProtocol.subscribeRun
-            this.state = TaskStates.Cancelled;
-        } else if (this.state !== TaskStates.Cancelled) {
+        if (self.state === TaskStates.Running) {
             // Task cancellation is passed down to the job first
             // and then bubbled up into the promise chain
             // created in this.run()
-            return this.job.cancel(error);
+            self.state = TaskStates.Cancelled;
+            return self.job
+                .cancel(error)
+                .catch(function(err) {
+                    logger.error('Swallow the Error when task cancelled', {
```

Comment:
This error is expected, since user cancel the job purposely, so use `logger.error` doesn't make sense.